pkg/version/version.go

```diff
@@ -63,40 +63,41 @@ func Get() version.Info {
 // 5. Overwrite that chunk of the bytes if found, otherwise return error.
 
 var (
-	// defaultReleaseInfoPadded may be replaced in the binary with a pull spec that overrides defaultReleaseInfo as
+	// defaultReleaseInfoPadded may be replaced in the binary with Release Metadata: Version that overrides defaultVersion as
 	// a null-terminated string within the allowed character length. This allows a distributor to override the payload
 	// location without having to rebuild the source.
-	defaultReleaseInfoPadded = "\x00_RELEASE_IMAGE_LOCATION_\x00XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\x00"
-	defaultReleaseInfoPrefix = "\x00_RELEASE_IMAGE_LOCATION_\x00"
-	defaultReleaseInfoLength = len(defaultReleaseInfoPadded)
+	defaultVersionPadded = "\x00_RELEASE_VERSION_LOCATION_\x00XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\x00"
+	defaultVersionPrefix = "\x00_RELEASE_VERSION_LOCATION_\x00"
+	defaultVersionLength = len(defaultVersionPadded)
 )
 
 // ExtractVersion() abstracts how the binary loads the default release payload. We want to lock the binary
-// to the pull spec of the payload we test it with, and since a payload contains the cli image we can't
-// know that at build time. Instead, we make it possible to replace the release string after build via a
+// to version of the payload we test it with, and since a payload contains the cli image we can't
+// know that at build time. Instead, we make it possible to replace the version string after build via a
 // known constant in the binary.  When extracting oc from release image, 'oc version' reports the payload version.
 func ExtractVersion() (version.Info, string, error) {
-	if strings.HasPrefix(defaultReleaseInfoPadded, defaultReleaseInfoPrefix) {
+	if strings.HasPrefix(defaultVersionPadded, defaultVersionPrefix) {
 		return Get(), "", nil
 	}
-	nullTerminator := strings.IndexByte(defaultReleaseInfoPadded, '\x00')
+	nullTerminator := strings.IndexByte(defaultVersionPadded, '\x00')
 	if nullTerminator == -1 {
 		// the binary has been altered, but we didn't find a null terminator within the release name constant which is an error
-		return version.Info{}, "", fmt.Errorf("release name location was replaced but without a null terminator before %d bytes", defaultReleaseInfoLength)
+		return Get(), "", fmt.Errorf("release name location was replaced but without a null terminator before %d bytes", defaultVersionLength)
```

Comment:
The oc binary doesn't depend on the version being pinned to the release, but if we're meddling w/ the binary and the result isn't what is expected, it should return an error? `oc` works the same either way, though, so I could see returning a warning for these, instead.  Also, setting version to the Git version info rather than nothing, in these.

Reply:
I can't figure this out, when I extract oc from the release image from this PR, I get an empty releaseName (match found, but len(releaseName)==0. However, when I create a release from a published release w/ new cli image (built from this PR, but locally), and extract oc from that, releaseName is the expected 4.3.0-0.ci-2019-10-23-blah - maybe the ci-op/release:lastes image is in flux w/ multiple pushes/test restarts so the Release Metadata: Version is absent? not sure.